File: web/teacher/src/types/store/lesson.ts
import { BaseSubject } from '../api/v1'
import { SchoolType } from './common'
import { Student, Teacher } from '.'

export interface Subject extends BaseSubject {
  schoolType: SchoolType
}

export interface Lesson {
  id: number
  shiftId: number
  subjectId: number
  room: number
  teacherId: string
  studentId: string
  startAt: string
  endAt: string
  notes: string
  createdAt: string
  updatedAt: string
}

export interface LessonState {
  subjects: Subject[]
  lessons: Lesson[]
  teachers: Teacher[]
  students: Student[]
}

export interface SubjectMap {
  [key: number]: Subject
}

export type SubjectsMap = {
  [key in SchoolType]: Subject[]
}

File: api/internal/gateway/teacher/v1/request/lesson.go
package request

type CreateLessonRequest struct {
	ShiftID   int64  `json:"shiftId,omitempty"`   // 授業スケジュールID
	SubjectID int64  `json:"subjectId,omitempty"` // 授業科目ID
	Room      int32  `json:"room,omitempty"`      // 教室番号
	TeacherID string `json:"teacherId,omitempty"` // 講師ID
	StudentID string `json:"studentId,omitempty"` // 生徒ID
}
